File: 4.ROS/Tasks/Task_1/nomeer_ws/src/pub_pkg/src/pub.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>
#include <sstream>

int main(int argc, char** argv)
{
    ros::init(argc, argv, "publisher");
    ros::NodeHandle nh;
    ros::Rate rate(1);

    ros::Publisher pub;
    pub = nh.advertise<std_msgs::String>("string_topic",1);

    std_msgs::String msg;
    std::stringstream ss;
    ss << "I am Learning Robotics and ROS";
    msg.data = ss.str();
    
    while(ros::ok())
    {
        pub.publish(msg);
        ros::spinOnce();
        rate.sleep();
    }

    return 0;
}

File: 4.ROS/Tasks/Task_1/nomeer_ws/src/pub_pkg/scripts/pub.py
#!/usr/bin/python3

import rospy
from std_msgs.msg import String

def Talker():
    rospy.init_node("talker")
    pub = rospy.Publisher("msg_topic", String, queue_size=1)
    rate = rospy.Rate(1)

    msg = "I am Learning Robotics and ROS"
    while not rospy.is_shutdown():
        pub.publish(msg)
        rate.sleep()

if __name__ == "__main__":
    try:
        Talker()
    except rospy.ROSInterruptException:
        pass